adam/batch_propagation_results.py

```diff
@@ -0,0 +1,271 @@
+        """Get the final positions of all propagated objects in the job.
+
+        Args:
+            position_orbit_type (PositionOrbitType): the type of orbit position to filter.
+            force_update (boolean): whether the request should be re-executed.
+
+        Returns:
+            list: A list of the final orbit positions, filtered by the position_orbit_type.
+        """
+
+        self.__update_results(force_update)
+        position_type_string = position_orbit_type.value
+        final_positions = self._detailedOutputs['finalPositionsByType'].get(position_type_string)
+        if final_positions is None:
+            return []
+        positions = final_positions['finalPosition']
+        return_data = list(
+            map(lambda p: [np.datetime64(dateparser.parse(p['epoch'])), p['x'], p['y'], p['z']],
+                positions))
+        return return_data
+
+    def get_result_ephemeris_count(self, force_update=False):
+        """Get the number of ephemerides.
+
+        Args:
+            force_update (boolean): whether the request should be re-executed.
+
+        Returns:
+            int: the number of ephemerides generated from the propagation.
+        """
+
+        self.__update_results(force_update)
+        ephemeris = self._detailedOutputs.get('ephemeris')
+        if ephemeris is None:
+            return 0
+        ephemeris_objects = ephemeris.get('ephemerisResourcePath')
+        if ephemeris_objects is None:
+            return 0
+        return len(ephemeris_objects)
+
+    def get_result_raw_ephemeris(self, run_number, force_update=False):
+        """Get an ephemeris for a particular run in the batch.
+
+        Args:
+            force_update (boolean): whether the request should be re-executed.
+
+        Returns:
+            str: the ephemeris file as a string.
+        """
+
+        self.__update_results(force_update)
+        ephemeris = self._detailedOutputs['ephemeris']
+        ephemeris_resource_name = ephemeris['ephemerisResourcePath'][run_number]
+        base_url = ephemeris['resourceBasePath']
+        if not base_url.endswith('/'):
+            base_url = base_url + '/'
+        url = urllib.parse.urljoin(base_url, ephemeris_resource_name)
+        with urllib.request.urlopen(url) as response:
+            return response.read().decode('utf-8')
+
+    def get_result_ephemeris(self, run_number, force_update=False):
+        """Get an ephemeris for a particular run in the batch as a Panda DataFrame
+
+        Args:
+            force_update (boolean): whether the request should be re-executed.
+
+        Returns:
+            ephemeris: Ephemeris from file as a Pandas DataFrame
+        """
+
+        ephemeris_text = self.get_result_raw_ephemeris(run_number, force_update)
+        ephemeris = stk.io.ephemeris_file_data_to_dataframe(ephemeris_text.splitlines())
+        return ephemeris
+
+    def __update_results(self, force_update):
```

Comment:
Is the double leading underscore needed (for name mangling), or would it be appropriate as a single underscore?

Reply:
Just a hold over typo from the original implementation I think